src/Tasks/Microsoft.NET.Build.Extensions.Tasks.UnitTests/GivenAGetDependsOnNETStandardTask.cs

```diff
@@ -0,0 +1,152 @@
+            }
+            finally
+            {
+                File.Delete(testFile);
+            }
+        }
+
+        [Fact]
+        public void ReturnsFalseForNativeLibrary()
+        {
+            var corelibLocation = typeof(object).GetTypeInfo().Assembly.Location;
+            var corelibFolder = Path.GetDirectoryName(corelibLocation);
+
+            // do our best to try and find it, intentionally only use the windows name since linux will not be a PE.
+            var coreclrLocation = Directory.EnumerateFiles(corelibFolder, "coreclr.dll").FirstOrDefault();
+
+            // if we can't find it, skip the test
+            if (coreclrLocation != null)
+            {
+                // ensure that false is returned and no warning is logged for native library
+                var task = new GetDependsOnNETStandard()
+                {
+                    BuildEngine = new MockBuildEngine(),
+                    References = new[] { new MockTaskItem() { ItemSpec = coreclrLocation } }
+                };
+
+                task.Execute().Should().BeTrue();
+                task.DependsOnNETStandard.Should().BeFalse();
+                // don't warn for PE with no metadata
+                ((MockBuildEngine)task.BuildEngine).Warnings.Count.Should().Be(0);
+            }
+        }
+
+        [Fact]
+        public void SucceedsOnMissingFileReturnsFalse()
+        {
+            var missingFile = $"{nameof(SucceedsOnMissingFileReturnsFalse)}.shouldNotExist.dll";
+            File.Exists(missingFile).Should().BeFalse();
+            var task = new GetDependsOnNETStandard()
+            {
+                BuildEngine = new MockBuildEngine(),
+                References = new[] { new MockTaskItem() { ItemSpec = missingFile } }
+            };
+
+            task.Execute().Should().BeTrue();
+            task.DependsOnNETStandard.Should().BeFalse();
+            ((MockBuildEngine)task.BuildEngine).Warnings.Count.Should().Be(0);
+        }
+
+        [Fact]
+        public void SuccedsWithWarningOnLockedFile()
+        {
+            var lockedFile = $"{nameof(SuccedsWithWarningOnLockedFile)}.dll";
+
+            try
+            {
+                var task = new GetDependsOnNETStandard()
+                {
+                    BuildEngine = new MockBuildEngine(),
+                    References = new[] { new MockTaskItem() { ItemSpec = lockedFile } }
+                };
+
+                // create file with no sharing
+                using (var fileHandle = new FileStream(lockedFile, FileMode.Create, FileAccess.ReadWrite, FileShare.None))
+                {
+                    task.Execute().Should().BeTrue();
+                    task.DependsOnNETStandard.Should().BeFalse();
+                    ((MockBuildEngine)task.BuildEngine).Warnings.Count.Should().BeGreaterThan(0);
+                }
+            }
+            finally
+            {
+                File.Delete(lockedFile);
+            }
+        }
+    }
```

Comment:
We have #1287 with on overview of the test scenarios we want.  I'd like to merge these changes as soon as they're ready so that we can work on the insertion into the CLI and VS in parallel while Eric works on those tests.